client/src/components/Auth/Login/Login.jsx

```diff
@@ -0,0 +1,112 @@
+import React, { Component } from 'react';
+import PropTypes from 'prop-types';
+import { withRouter, Link } from 'react-router-dom';
+import { connect } from 'react-redux';
+
+import { loginUser } from '../../../actions/authActions';
+import { Container, Header, TopText, ContentArea, Footer } from '../Auth.styled';
+import FormGroup from '../../Shared/Forms/FormGroup';
+import { Button } from '../../Shared/Forms/Form.styled';
+
+class Login extends Component {
+	state = {
+		email: '',
+		password: '',
+		errors: {},
+	};
+
+	static propTypes = {
+		loginUser: PropTypes.func.isRequired,
+		errors: PropTypes.shape({
+			email: PropTypes.string,
+			password: PropTypes.string,
+		}).isRequired,
+		auth: PropTypes.shape({
+			isAuthenticated: PropTypes.bool.isRequired,
+			user: PropTypes.object.isRequired,
+		}).isRequired,
+	};
+
+	componentWillMount() {
+		const { auth, history } = this.props;
+		if (auth.isAuthenticated) history.push('/dashboard');
+	}
+
+	componentWillReceiveProps(nextProps) {
+		if (nextProps.errors) {
+			this.setState({ errors: nextProps.errors });
+		}
+	}
+
+	handleChange = e => {
+		const { name, value } = e.target;
+		this.setState({ [name]: value });
+	};
+
+	handleSubmit = e => {
+		e.preventDefault();
+		// eslint-disable-next-line no-shadow
+		const { loginUser, history } = this.props;
```

Comment:
Another quick note: We had discussion at my office about destructuring, because many of us used destructuring all the time, even if it's not needed that much, we decided in the end that we should use destructuring only if we have a lot of props that we're going to use in code, or if we want to destructure some inner properties from provided props. Sometimes, when you have a lot of code and a lot of variables, you'll want to know from where you're receiving those values (state, props, or something else).

Just wanted to share my experience regarding this here, this looks okay 👍

Reply:
Airbnb's strict linting rules and always needing to destructure seems like overkill a lot of the time. Your office's conclusion sounds reasonable 😀